### ISSUE TITLE
Cannot use `%{lib:...}` in the `flags` stanza.

### ISSUE THREAD

**Issue by samoht** [ocaml/dune#3362]
When using:

```
(library
  (name foo)
  (modules foo)
  (flags -cclib -I%{lib:foo:bar})
```

I’ve got:
```
Error: %{lib:..} isn't allowed in this position
```

However it works if I we use an intermediate file:

```
(library
  (name foo)
  (modules foo)
  (flags (:include toto)))
(rule
  (action (with-stdout-to toto (run echo -I%{lib:foo:bar}))))
```

**Comment by diml** [comment]
Indeed, we don't want people to do that. You'd be bypassing dune and not telling it that the C compiler is going to look inside `%{lib:foo:bar}`. This could lead to invalid builds.

There is however a clean way to do that, but I don't remember the syntax. /cc @snowleopard and @aalekseyev who worked on this.

**Comment by aalekseyev** [comment]
Actually I think we did not add any explicit syntax to add an include path to -I.
Dune just implicitly adds all source directories of you library dependencies to -I.
So, one reasonable (perhaps) workaround could be to define a dummy library foo_bar in the directory foo/bar and depend on it.

**Comment by robinbb** [comment]
@samoht This appears to be fixed. The error message `%{lib:..} isn't allowed in this position` is no longer present in the codebase, and `%{lib:...}` now expands successfully in `(flags)` stanzas.

~Could you close this if you agree?~
